### ISSUE TITLE
OutOfMemoryError in BitmapHelper.getColoredBitmap() 

### ISSUE THREAD

**Issue by urizev** [TheFinestArtist/FinestWebView-Android#38]
Hi everyone,

I don't know if anyone more have received this error:

```
Fatal Exception: java.lang.OutOfMemoryError: Failed to allocate a 82956 byte allocation with 48202 free bytes and 47KB until OOM
       at dalvik.system.VMRuntime.newNonMovableArray(VMRuntime.java)
       at android.graphics.Bitmap.nativeCopy(Bitmap.java)
       at android.graphics.Bitmap.copy(Bitmap.java:680)
       at com.thefinestartist.finestwebview.helpers.BitmapHelper.getColoredBitmap(BitmapHelper.java:31)
       at com.thefinestartist.finestwebview.helpers.BitmapHelper.getColoredBitmap(BitmapHelper.java:38)
       at com.thefinestartist.finestwebview.FinestWebViewActivity.updateIcon(FinestWebViewActivity.java:834)
       at com.thefinestartist.finestwebview.FinestWebViewActivity.layoutViews(FinestWebViewActivity.java:461)
       at com.thefinestartist.finestwebview.FinestWebViewActivity.onCreate(FinestWebViewActivity.java:853)
       at android.app.Activity.performCreate(Activity.java:6289)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2655)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2767)
       at android.app.ActivityThread.access$900(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1449)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5951)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
```

This is the way I launching the web activity:

```
public static void openUrlEmbedded(Context context, String url, String title) {
     FinestWebView.Builder builder;
     builder = new FinestWebView.Builder(context)
                .titleColor(Color.WHITE)
                .urlColor(Color.WHITE)
                .iconDefaultColor(Color.WHITE)
                .showIconMenu(false)
                .showIconForward(false)
                .showIconBack(false)
                .backPressToClose(false);

    if (title != null) {
         builder.titleDefault(title);
    }

    builder.show(url);
}
```

It is a pretty simple code. 

I cannot say which url is throwing the error since the app is in production. If it really matters I could try to find out the url but I am not sure I could.

Thanks in advance.

**Comment by TheFinestArtist** [comment]
Hello @urizev, thanks for your bug report! I recently updated 4 icons that was used in `FinestWebView`. `BitmapHelper` no longer have `getColoredBitmap()` method. Since VectorDrawableCompat is added newly released support package, I updated all the drawable as vector drawable. I think `OutOfMemoryError` will be gone after you update `FinestWebView` to `1.2.1`.

Let me know if `OutOfMemoryError` keep occurs!
